lib/spack/spack/installer.py

```diff
@@ -814,6 +814,13 @@ def _prepare_for_install(self, task):
         # Determine if the spec is flagged as installed in the database
         rec, installed_in_db = self._check_db(task.pkg.spec)
 
+        # Check for an install prefix collision
+        available = spack.store.db.is_occupied_install_prefix(task.pkg.spec.prefix)
```

Comment:
(question, no change required) I think this could have also been fixed by placing a call to `check_installed` here. I'm curious how much this speeds things up for you (I imagine if you frequently reinstall large DAGs where many specs are already installed, this could consume some time for each already-installed spec).

Reply:
Yeah, I was trying to avoid the latter, see the asciinema links above and imagine the last spec is being dev'ed and reinstalled every now and then.